src/app/sessions/services/sessions-statistics-service.spec.ts

```diff
@@ -0,0 +1,141 @@
+import { SessionsStatisticsService } from './sessions-statistics-service';
+import { SessionViewModel } from '../models/session.viewmodel';
+import * as moment from 'moment';
+import { HearingPart } from '../../hearing-part/models/hearing-part';
+
+let sessionsStatisticsService: SessionsStatisticsService;
+let session: SessionViewModel;
+
+describe('SessionsStatisticsService', () => {
+    beforeAll(() => {
+        sessionsStatisticsService = new SessionsStatisticsService();
+    });
+
+    describe('calculateAllocatedHearingsDuration', () => {
+        it('should give one hearing part duration given session with one hearing', () => {
+            session = {
```

Comment:
Can we create even a private method in this test getEmptySession that returns a session with all fields as undefined and then use the spread operator to fill only the required fields?

let session = { ...this.getEmptySession(), duration: 10, hearingParts: {[...this.getEmptyHearingPart(), duration: moment.duration(PT20M)}]}

Reply:
Or even go for a separate Builders